### ISSUE TITLE
Error not propagating properly in Polars 26.1

### ISSUE THREAD

**Issue by AnatolyBuga** [pola-rs/polars#6097]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Issue description

Hello Polars Team,

After upgrading to Polars **26.1** from 25.1 some issues came up:
1) The **error message is misleading**. You will get 
```
ComputeError(
        Owned(
            "returned aggregation is a different length: 0 than the group lengths: 2",
        ),
    ),
)
```
The **original** error however is this, that is what should be returned to the user:
```
ComputeError(
        Owned(
            ""Some Error"",
        ),
    ),
)
```

2) **This error wouldn't even come up in 25.1**  Below code worked well in 25.1 but not in 26.1. Now that I know what the issue is I've put a workaround, yet see if you consider this as a bug.

### Reproducible example

```rust
use polars::prelude::*;

pub fn example() {
   let df = df![
            "rf" => ["App", "App", "Gg", "App"],
            "x" => ["Hey", "There", "Ante", "R"],
            "y" => [Some(-1.11), Some(2.),None, Some(3.4)],
            "z" => [Some(-1.11), Some(2.),None, Some(3.4)],
        ].unwrap();
    let res = df.lazy()
        .with_streaming(false)
        .groupby_stable([col("rf")])
        .agg([issue().alias("RESULT")])
        .collect();
    dbg!(res);
}

fn issue() -> Expr {
    apply_multiple(
        move |columns| {
        return Err(PolarsError::ComputeError("Some Error".into()))
        },
        &[
            col("x"),
            col("y")
        ],
        GetOutput::from_type(DataType::Float64),
        true
    )
}
```

<details>
polars = {version="0.26.1", features=["lazy"]}
</details>

**Comment by phaile2** [comment]
Hello I took a glance at this since it was marked `good first issue` It looks as though the Result is converted directly to a Some [here](https://github.com/pola-rs/polars/blob/master/polars/polars-lazy/src/physical_plan/expressions/apply.rs#L287), so the Error doesn't get propagated. Should this function instead return an Err if `call_udf` fails?

**Comment by phaile2** [comment]
I think this was fixed in this [PR](https://github.com/pola-rs/polars/pull/6629)

**Comment by AnatolyBuga** [comment]
@phaile2 , yes indeed you are right! Amazing. I might as well PR a unit test for that. What do you think?

**Comment by AnatolyBuga** [comment]
fixed in #6629

**Comment by AnatolyBuga** [comment]
fixed in https://github.com/pola-rs/polars/pull/6629
